packages/office-addin-mock/src/officeMockObject.ts

```diff
@@ -0,0 +1,179 @@
+import { usageDataObject } from "./defaults";
+
+/**
+ * Creates an office-js mockable object
+ * @param object Object structure to provide initial values for the mock object (Optional)
+ */
+export class OfficeMockObject {
+  constructor(object?: ObjectData) {
+    this.properties = new Map<string, OfficeMockObject>();
+    this.loaded = false;
+    this.resetValue(undefined);
+    if (object) {
+      this.populate(object);
+    }
+  }
+
+  /**
+   * Adds a function to OfficeMockObject. The function can be accessed by simply calling `this.methodName`
+   * @param methodName Method name of the function to be added
+   * @param methodBody Function to be added to the object. A blank function will be added if no argument is provided (Optional)
+   */
+  addMockFunction(methodName: string, methodBody?: Function) {
+    this[methodName] = methodBody ? methodBody : function () {};
+  }
+
+  /**
+   * Adds a mock object to OfficeMockObject. The object can be accessed by simply calling `this.objectName`
+   * @param objectName Object name of the object to be added
+   */
+  addMock(objectName: string) {
+    const officeMockObject = new OfficeMockObject();
+    officeMockObject.isTreatedLikeObject = true;
+    this.properties.set(objectName, officeMockObject);
+    this[objectName] = this.properties.get(objectName);
+  }
+
+  /**
+   * Mock replacement of the load method in the Office.js API
+   * @param propertyArgument Argument of the load call. Will load any properties in the argument
+   */
+  load(propertyArgument: string | string[]) {
```

Comment:
Is it possible for the load call to have multiple arguments?  How would we account for a variable list of arguments?

Reply:
It can have just a string or an array of strings. I am accounting for that.
If it is an array, it will do a `forEach` on each of the properties. If is just a string, we simply do a normal load on it.

Reply:
The linter rule code that parses the load method is looking for multiple method arguments instead of just one . . . so I'm not sure which one is accurate.

Reply:
What do mean about multiple method arguments? They should be the same

Reply:
In the linter rule's load.ts "parseLoadArguments(node: TSESTree.MemberExpression)" method the line "node.parent.arguments.forEach" looks for for more then just one argument.  So the linter rule is looking for multiple arguments but this mock code is not.

Reply:
Probably you are mistaken in some outdated code PR version. This is how the code is right now:
```
    if (typeof propertyArgument === "string") {
      this.loadMultipleProperties(propertyArgument);
    } else {
      propertyArgument.forEach((property: string) => {
        this.loadMultipleProperties(property);
      });
```
There is a forEach here.

Reply:
That foreach is looking at at each property in an array of string that came from a single method argument.  In the linter rule you are looking at multiple method arguments (and after looking more closely at the linter rule I don't think we're looking for a string array as an argument)

Reply:
Both cases should be looking at receiving one string that represents multiples properties: `load("address, font");` or multiple arguments: `load("address", "font");`. There are test cases for both of them, so they are covered. 

In the linter rules:
```
		{
			code: `
			  var range = worksheet.getSelectedRange();
			  range.load("font/fill/color, address");
			  await context.sync();
			  console.log(range.font.fill.color);
			  console.log(range.address);`
		},
		{
			code: `
			  var range = worksheet.getSelectedRange();
			  range.load("font/fill/color", "address");
			  await context.sync();
			  console.log(range.font.fill.color);
			  console.log(range.address);`
		},
```

In the mocking:
```
    it("Multiple properties load", async function() {
      const officeMock = new OfficeMockObject(testObject);
      officeMock.load("range/color, range/font/size");
      officeMock.sync();
      assert.strictEqual(officeMock.range.getColor(), "blue");
      assert.strictEqual(officeMock.range.font.size, 12);
    });
    it("Comma separated load", async function() {
      const officeMock = new OfficeMockObject(testObject);
      officeMock.load(["range/color", "range/font/type", "range/font/size"]);
      officeMock.sync();
      assert.strictEqual(officeMock.range.getColor(), "blue");
      assert.strictEqual(officeMock.range.font.type, "arial");
      assert.strictEqual(officeMock.range.font.size, 12);
    });
```

Reply:
Your test's for the linter and mocking are different and pass because they are checking for what they are coded for.
'load(["font/fill/color", "address"]);' is not the same as 'load("font/fill/color", "address");'

Add this test to your mock:
    it("Multi argument load", async function() {
      const officeMock = new OfficeMockObject(testObject);
      officeMock.load("range/color", "range/font/type", "range/font/size");
      officeMock.sync();
      assert.strictEqual(officeMock.range.getColor(), "blue");
      assert.strictEqual(officeMock.range.font.type, "arial");
      assert.strictEqual(officeMock.range.font.size, 12);
    });

And this test to the linter:
		{
			code: `
			  var range = worksheet.getSelectedRange();
			  range.load(["font/fill/color", "address"]);
			  await context.sync();
			  console.log(range.font.fill.color);
			  console.log(range.address);`
		},

I suspect they will both fail.

Reply:
Thanks for noticing that. The eslint one is wrong, we can't have multiple load arguments like that. I will work on that on the eslint thread. The mocking is the correct way to load, so we can proceed with this PR